driver-core/src/main/com/mongodb/internal/connection/SocketStream.java

```diff
@@ -240,7 +245,7 @@ public void close() {
             if (socket != null) {
                 socket.close();
             }
-        } catch (IOException e) {
+        } catch (IOException | RuntimeException e) {
```

Comment:
We have another instance where the socket is closed [KeyManagementService.java](https://github.com/mongodb/mongo-java-driver/blob/b60f8723b453d88ab14df69d1e2cfc18f09b9887/driver-sync/src/main/com/mongodb/client/internal/KeyManagementService.java#L101). Should we also implement a defensive measure here as well?

Reply:
Found this done in another [PR](https://github.com/mongodb/mongo-java-driver/pull/1204/files).